packages/playground/website/public/logger.php

```diff
@@ -4,6 +4,22 @@
 
 header('Content-Type: application/json');
 
+/**
+ * Rate limit logger requests
+ */
+$max_requests_per_hour = 5;
+
+session_start();
+$rate_limit_key = md5('logger_requests_' . date('Y-m-d H'));
+if (!isset($_SESSION[$rate_limit_key])) {
+    $_SESSION[$rate_limit_key] = 0;
+}
+$_SESSION[$rate_limit_key]++;
+
+if ($_SESSION[$rate_limit_key] > $max_requests_per_hour) {
+    response(false, 'Too many requests');
```

Comment:
would it be helpful to send a 429 here instead of what I presume is a 200?

also, these `response()` responses seem quite sparse. maybe we could help clue-in the browsers to what is happening?

```php
header( 'Content-type: application/json', true, 429 );
```

Do we expect the clients to heed this response and wait to send new logs?

Reply:
This API is currently used only by one modal and it shows the same error message whenever a request fails, so it's not needed today.

If the API gets used more in the future, using the correct response code and improving errors would be good.

> Do we expect the clients to heed this response and wait to send new logs?

No, users might click on the form again and if it fails, it would result in the same error.
Thinking about it now, the UX isn't great, but let's see how much it will be used before investing more time into it.